src/pgtz/build.rs

```diff
@@ -176,66 +178,106 @@ fn main() -> Result<()> {
+                "('{abbrev}', {pg_sql_utc_offset}, {pg_sql_is_dst})"
+            ));
+            mz_abbrev_buf.writeln(format!(
+                "('{abbrev}', {mz_sql_fixed_offset}, {mz_sql_is_dst}, {mz_sql_tz_name})"
+            ));
 
             phf_map.entry(UncasedStr::new(abbrev), abbrev);
 
             emitted_abbrev = true;
         }
 
-        sql_buf.end_line();
+        pg_abbrev_buf.end_line();
+        mz_abbrev_buf.end_line();
 
         rust_buf.writeln(format!(
             "pub static TIMEZONE_ABBREVS: phf::Map<&'static UncasedStr, TimezoneAbbrev> = {};",
             phf_map.build(),
         ));
 
-        fs::write(out_dir.join("abbrev.gen.sql"), sql_buf.into_string())?;
+        fs::write(out_dir.join("abbrev.gen.sql"), pg_abbrev_buf.into_string())?;
+        fs::write(
+            out_dir.join("mz_abbrev.gen.sql"),
+            mz_abbrev_buf.into_string(),
+        )?;
         fs::write(out_dir.join("abbrev.gen.rs"), rust_buf.into_string())?;
     }
 
     // Convert chrono-tz's list of timezones into the SQL definition of the
-    // pg_timezone_names view.
+    // pg_timezone_names and mz_timezone_names views.
     {
-        let mut sql_buf = CodegenBuf::new();
+        let mut pg_sql_buf = CodegenBuf::new();
+        let mut mz_sql_buf = CodegenBuf::new();
 
-        sql_buf.writeln("VALUES");
+        pg_sql_buf.writeln("VALUES");
+        mz_sql_buf.writeln("VALUES");
 
         for (i, tz) in TZ_VARIANTS.iter().enumerate() {
             let name = tz.name();
             if i > 0 {
-                sql_buf.writeln(",");
+                pg_sql_buf.writeln(",");
+                mz_sql_buf.writeln(",");
             }
-            sql_buf.write("(");
-            sql_buf.write(format!("'{name}',"));
-            sql_buf.write(format!("timezone_offset('{name}', now()).abbrev,"));
-            sql_buf.write(format!("timezone_offset('{name}', now()).base_utc_offset + timezone_offset('{name}', now()).dst_offset,"));
-            sql_buf.write(format!(
+            pg_sql_buf.write("(");
+            pg_sql_buf.write(format!("'{name}',"));
+            pg_sql_buf.write(format!("timezone_offset('{name}', now()).abbrev,"));
+            pg_sql_buf.write(format!("timezone_offset('{name}', now()).base_utc_offset + timezone_offset('{name}', now()).dst_offset,"));
+            pg_sql_buf.write(format!(
                 "timezone_offset('{name}', now()).dst_offset <> interval '0'"
             ));
-            sql_buf.write(")");
+            pg_sql_buf.write(")");
+
+            mz_sql_buf.write("(");
+            mz_sql_buf.write(format!("'{name}'"));
+            mz_sql_buf.write(")");
         }
 
-        fs::write(out_dir.join("timezone.gen.sql"), sql_buf.into_string())?;
+        fs::write(out_dir.join("timezone.gen.sql"), pg_sql_buf.into_string())?;
+        fs::write(
+            out_dir.join("mz_timezone.gen.sql"),
+            mz_sql_buf.into_string(),
+        )?;
     }
```

Comment:
To minimize the amount of codegen, now that we've got the `mz_catalog` tables, I think it'd be great to _stop_ codegening the `pg_catalog` tables and instead define them in terms of the new `mz_catalog` tables. Will just be easier to follow.

Reply:
Agree. Made pg_ views over mz_ which resulted in moving much of the generated code into builtin.rs.